### ISSUE TITLE
We found conflicting component dependencies 

### ISSUE THREAD

**Issue by kalalruturaj** [magento/magento2#20684]
I have upgraded my Magento from 2.16 to Magento 2.3.0 and since then I am not able to install any extension. I am keep getting an error message that Check Component Dependency. I have also attached the composer.json file in the bottom section

 I have refereed  all of the gift-hub tickets             
         1. [#7917](https://github.com/magento/magento2/issues/7917)
         2. [#19850](https://github.com/magento/magento2/issues/19850)
         3. [#3207](https://github.com/magento/magento2/issues/3207)
### Preconditions (*)

1. Magento 2.3.0 not able to pass the Read Link check, Component Dependency erros
2. Php 7.2, Php-fpm-7.2
3. I also Referred this docs as well [Click Here for Docs](https://devdocs.magento.com/guides/v2.3/comp-mgr/trouble/cman/component-depend.html)
4. This is composer.json file [Click Here](https://paper.dropbox.com/doc/Composer.json--AWcUTeSbS2LlB~0czLFrP99jAQ-J8XIdhc7bRgsEa3Wce83P)
### Steps to reproduce (*)

1. web setup wizard -> Extension manager -> Review and Install
2. Select paypal/module-braintree V3.0.6 and Click Install -> Readiness Check
![image](https://user-images.githubusercontent.com/14205802/51815289-d7eedf00-228d-11e9-8ada-53f921a4115a.png)

This is the PayPal BrainTree Magento 2.3.0 Version extension -> Here is the [link Click](https://marketplace.magento.com/paypal-module-braintree.html)

### Expected result (*)

1. Expected to Install extension on store
2.

### Actual result (*)

1. ![image](https://user-images.githubusercontent.com/14205802/51815289-d7eedf00-228d-11e9-8ada-53f921a4115a.png)
2. This is the error message what i get. 
![image](https://user-images.githubusercontent.com/14205802/51815467-1afd8200-228f-11e9-8fd0-389b373c3091.png)

Check Component Dependency
We found conflicting component dependencies. Hide detail

For additional assistance, see component dependency help .

**Comment by magento-engcom-team** [comment]
Hi @kalalruturaj. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@kalalruturaj do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by kalalruturaj** [comment]
@magento-engcom-team give me 2.3.0  instance

**Comment by magento-engcom-team** [comment]
Hi @kalalruturaj. Thank you for your request. I'm working on Magento 2.3.0  instance for you

**Comment by magento-engcom-team** [comment]
Hi @milindsingh. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
Hi @milindsingh. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by kalalruturaj** [comment]
Does anyone have any solutions for this ?

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-97963` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by dandelionred** [comment]
@kalalruturaj Have you tried raising memory_limit for php? In my case the 756M stock memory limit in the stock magento configs (both apache and nginx ones) was the problem. Try raising it to 2G as the docs recommend for setup tasks.

https://github.com/magento/magento2/issues/7917#issuecomment-453658759

**Comment by TechnoBuddhist** [comment]
@dandelionred Thank you, your link pointed me in the right direction to resolve my issue! I raised memory_limits in .htaccess and .user.ini to 2G from 756M.

**Comment by dsaha1656** [comment]
> @dandelionred Thank you, your link pointed me in the right direction to resolve my issue! I raised memory_limits in .htaccess and .user.ini to 2G from 756M.

this one also worked for me,

**Comment by soopermouse** [comment]
that didn't work for me.
The fact the error message doesn't give any useful information IS the issue.
Why are we still using Magento?

**Comment by mwhembree** [comment]
Raising memory limits always works for me. However, every upgrade overwrites the .htaccess and .user.ini files with the old 756M value and I have to change the memory limits again to install any extensions. Since 756M is no good, why not change the default to 2G?

**Comment by dandelionred** [comment]
@mwharron Just prepare a patch for those two files and apply it manually right after upgrading.

> Since 756M is no good, why not change the default to 2G?

Because it is the community edition I guess.

**Comment by shahbaz-anjum** [comment]
Hi, following solution worked for me.

First open the magento 2 composer.json file and remove following lines of code.
```
"require-dev": {
        "allure-framework/allure-phpunit": "~1.2.0",
        "dealerdirect/phpcodesniffer-composer-installer": "^0.5.0",
        "friendsofphp/php-cs-fixer": "~2.14.0",
        "lusitanian/oauth": "~0.8.10",
        "magento/magento-coding-standard": "~4.0.0",
        "magento/magento2-functional-testing-framework": "2.5.3",
        "pdepend/pdepend": "2.5.2",
        "phpcompatibility/php-compatibility": "^9.3",
        "phpmd/phpmd": "@stable",
        "phpunit/phpunit": "~6.5.0",
        "sebastian/phpcpd": "~3.0.0",
        "squizlabs/php_codesniffer": "~3.4.0"
    }
```

After that set the memory_limit in php.ini to 2G
`memory_limit = 2G`
Updated the magento 2 .htaccess file also as this overwrites the memory_limit configuration set in php.ini, so update the `php_value memory_limit 2048M` in two locations in .htaccess file also.
After that try again Readiness Check this will resolve the issue.

**Comment by dandelionred** [comment]
@shahbaz-anjum 
> First open the magento 2 composer.json file and remove following lines of code.

Dev dependencies are not installed by default. So this change doesn't affect anything.

**Comment by gioamato** [comment]
I can confirm that raising memory_limits in .htaccess and .user.ini to 2G is still required in v2.3.4.

I don't think that the sample files should edited. The Web Setup Wizard should check the php environment configuration during readiness check.

**Comment by magento-engcom-team** [comment]
Hi @kalalruturaj. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-24335` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.
